addons/udes_sale_stock/models/sale_order_line.py

```diff
@@ -134,3 +135,35 @@ def _set_ui_is_cancelled(self):
             )
 
         lines_to_cancel.action_cancel()
+
+    def write(self, values):
+        """ If quantity is reduced, stock moves are not updated for them (but warning
+        issued). Update linked stock moves if those are still editable.
+        """
+        lines_qty_reduced  = self.env["sale.order.line"]
+        moves_editable = self.env["stock.move"]
+
+        if 'product_uom_qty' in values:
+            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+            moves_editable = self.move_ids.filtered(
+                lambda m: m.state not in ("draft", "done", "cancel")
+            ).sorted()
+            # Lines with quantity reduced and linked moves still editable
+            lines_qty_reduced = self.filtered(
```

Comment:
It sounds confusing to me,
I would suggest to have a method which returns lambda with a doc string.
```suggestion
            lines_qty_reduced = self.filtered(lines_qty_reduced_lambda)
```